openproblems/tasks/chromatin_potential/metrics/correlation.py

```diff
@@ -3,33 +3,45 @@
 import numpy as np
 import scipy.sparse
 import scipy.stats
+from sklearn.metrics import mean_squared_error
 
 
-def _correlation(adata, method="pearson"):
+def _metric(adata, method="pearson"):
     if method == "pearson":
         method = scipy.stats.pearsonr
-    else:
+    elif method == "spearman":
         method = scipy.stats.spearmanr
+    elif method == "mse":
+        method = mean_squared_error
 
-    cors = []
+    metrics = []
     for i in range(adata.shape[0]):
         x = (
             adata.obsm["gene_score"][i].toarray()[0]
             if scipy.sparse.issparse(adata.obsm["gene_score"])
             else adata.obsm["gene_score"][i]
         )
         y = adata.X[i].toarray()[0] if scipy.sparse.issparse(adata.X) else adata.X[i]
-        cors.append(method(x, y)[0])
-    cors = np.array(cors)
-    adata.obs["atac_rna_cor"] = cors
-    return np.median(cors[~np.isnan(cors)])
+        res = method(x, y)
+        if not isinstance(res, np.float64):
+            metrics.append(res[0])
```

Comment:
This doesn't make much sense to me. What's going on here?

Reply:
It looks like `scipy.stats.pearsonr` and `spearmanr` both return and effect size and a p value. Can you define this modification in the function? i.e. instead of using the scipy function at https://github.com/singlecellopenproblems/SingleCellOpenProblems/pull/139/files/0e7dabc9c22a30364059ca5f13efe057e8f036a5#diff-721ea0243e29a5e12a5c1b255ff562a01dc573e7134cd4934a766636299f9b4aR11 instead use 
```
def _pearsonr(x, y):
    return scipy.stats.pearsonr(x, y)[0]
```